### PR TITLE
feat(compaction): optimize for large number of small files

### PR DESCRIPTION
**Pull request by Little-Wallace**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## Checklist

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [ ] I have added test labels as necessary. See [details](https://github.com/risingwavelabs/risingwave/blob/main/docs/developer-guide.md#ci-labels-guide).
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [ ] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

- [ ] My PR contains critical fixes that are necessary to be merged into the latest release. (Please check out the [details](https://github.com/risingwavelabs/risingwave/blob/main/CONTRIBUTING.md))

## Documentation

- [ ] My PR needs documentation updates. (Please use the **Release note** section below to summarize the impact on users)

## Release note

If this PR includes changes that directly affect users or other significant modifications relevant to the community, kindly draft a release note to provide a concise summary of these changes. Please prioritize highlighting the impact these changes will have on users.

</details>

### CODE REVIEW

**Comment by Little-Wallace** [comment]
this branch: 
image

main branch:
image

File: src/meta/src/hummock/compaction/mod.rs [hunk 1]
```diff
@@ -191,14 +191,26 @@ pub fn create_compaction_task(
...
-        compaction_config.target_file_size_base
-    } else if input.target_level == base_level {
-        // This is just a temporary optimization measure. We hope to reduce the size of SST as much
-        // as possible to reduce the amount of data blocked by a single task during compaction,
-        // but too many files will increase computing overhead.
-        // TODO: remove it after can reduce configuration `target_file_size_base`.
-        compaction_config.target_file_size_base / 4
+    let target_file_size = if input.target_level == 0 || input.target_level == base_level {
+        let task_size = input.target_input_size + input.select_input_size;
+        if input.target_level > 0 && task_size < compaction_config.max_bytes_for_level_base {
```
Comment (Li0k) [inline src/meta/src/hummock/compaction/mod.rs:196]: IMO, It is sporadic that task size is smaller than max_bytes_for_level_base. So the original behavior is changed here.

my suggestion is
1. It is necessary to test daily nexmark and compare the performance.

2. Cover through UT to ensure that the output meets expectations

3. Trigger the branch through the special test mentioned in the PR.

**Comment by Little-Wallace** [comment]
I run nexmark q0~14 here: https://buildkite.com/risingwave-test/nexmark-benchmark/builds/3641
I run nexmark q14~105 here: https://buildkite.com/risingwave-test/nexmark-benchmark/builds/3640

**Comment by Little-Wallace** [comment]
This PR has been solved by https://github.com/risingwavelabs/risingwave/pull/17022

**Pull request closed without merging**